### PR TITLE
Main single positional

### PR DESCRIPTION
**Pull request by peschwa**
This PR brings handling of arguments to the programm more in line with what Getopt::Long would do. I.e. a parameter that is declared as Scalar (like in «sub MAIN(:$foo) { ... }») cannot be dispatched to by multiple occurences of a command line switch of the same name.

### CODE REVIEW

File: src/core/Main.pm [hunk 1]
```diff
@@ -150,7 +150,37 @@ my sub MAIN_HELPER($retval = 0) is hidden_from_backtrace {
     @matching_candidates .=grep: {!has-unexpected-named-arguments($_.signature, $n)};
     # If there are still some candidates left, try to dispatch to MAIN
     if +@matching_candidates {
-        return $m(|@($p), |%($n));
+        # verify that we don't dispatch multiple occurences of a command line argument
+        # to a parameter that is a Scalar
+        if %($n).grep: { .value ~~ Positional } -> @pairs {
+            for @pairs -> $pair {
+                for @matching_candidates -> $cand {
+                    my @param = $cand.signature.params.grep: { .named_names eq $pair.key };
```
Comment (Mouq) [inline src/core/Main.pm]: `.named_names` returns an array. E.g.:
```
sub foo(:name1(:name2(:$name3))) {}
say &foo.signature.params[0].named_names.perl
```
outputs `Array.new("name3", "name2", "name1")`

File: src/core/Main.pm [hunk 2]
```diff
@@ -150,7 +150,37 @@ my sub MAIN_HELPER($retval = 0) is hidden_from_backtrace {
     @matching_candidates .=grep: {!has-unexpected-named-arguments($_.signature, $n)};
     # If there are still some candidates left, try to dispatch to MAIN
     if +@matching_candidates {
-        return $m(|@($p), |%($n));
+        # verify that we don't dispatch multiple occurences of a command line argument
+        # to a parameter that is a Scalar
```
Comment (Mouq) [inline src/core/Main.pm]: My thought would be that we can still dispatch to them, but should tie-break. So `sub MAIN(:$arg) {}` could still receive `program --arg=1 --arg=2`, but `multi MAIN(:@arg) {}` beats `multi MAIN(:$arg) {}` with that invocation… OTOH, invocations like `-n=1,2` are still NYI, which in my mind is different than `-n=1 -n=2`, and there doesn't seem to be spec on the latter.

Comment (peschwa) [inline src/core/Main.pm]: I'm not fully satisfied with how this PR solves the dispatch, and as you say there's still insufficient speculation on some cases.

**Comment by peschwa** [comment]
Closing this for now.  I don't think the approach is the best, and as Mouq mentions there's a few cases that have to be speculated about first anyway.

**Pull request closed without merging**